.github/workflows/deploy-vue-storefront-cloud.yml

```diff
@@ -46,64 +46,27 @@ jobs:
-
-      - name: Deploy middleware on ${{ secrets.PROJECT_NAME }}.${{ secrets.CLOUD_REGION }}.gcp.storefrontcloud.io/api/
-        run: |
-          if curl -f -s -H 'X-User-Id: ${{ secrets.CLOUD_USERNAME }}' -H 'X-Api-Key: ${{ secrets.CLOUD_PASSWORD }}' -H 'Content-Type: application/json' -X PATCH -d '{
-            "front_version":"${{ github.sha }}",
-            "additional_apps": {
-              "apps":[{
-                "name":"middleware",
-                "tag":"${{ github.sha }}",
-                "image":"${{ secrets.DOCKER_REGISTRY_URL || 'registry.vuestorefront.cloud' }}/${{ secrets.PROJECT_NAME }}-storefrontcloud-io/vue-storefront-middleware",
-                "path":"/api/",
-                "port":"3000",
-                "has_base_path":false
-              }],
-              "enabled": true
-              }
-            }' ${{ secrets.CLOUD_API_BASE_URL || 'https://farmer.vuestorefront.cloud' }}/instance/${{ secrets.PROJECT_NAME }}-${{ secrets.CLOUD_REGION }}-gcp-storefrontcloud-io; then
-            echo "Instance updated"
-          else
-            echo "Something went wrong during the update process..."
-            exit 1
-          fi
-
-      - name: Update deployment status (success)
-        if: success()
-        uses: chrnorm/deployment-status@releases/v1
-        with:
-          token: "${{ github.token }}"
-          target_url: https://${{ secrets.PROJECT_NAME }}.${{ secrets.CLOUD_REGION }}.gcp.storefrontcloud.io
-          state: "success"
-          description: Congratulations! The deploy is done.
-          deployment_id: ${{ steps.deployment.outputs.deployment_id }}
+    - name: Deploy
+      uses: fjogeleit/http-request-action@v1
+      with:
+        url: '${{ secrets.CLOUD_API_BASE_URL }}/cloud/instances/${{ secrets.PROJECT_NAME }}-${{ secrets.CLOUD_REGION }}-gcp-storefrontcloud-io/deploy'
+        method: 'PATCH'
+        customHeaders: '{"Content-Type":"application/json"}'
+        data: '{ "cloudUserId": "${{ secrets.CLOUD_USERNAME }}", "cloudUserPassword": "${{ secrets.CLOUD_PASSWORD }}", "dockerImageHash": "${{ github.sha }}" }'
 
-      - name: Update deployment status (failure)
-        if: failure()
-        uses: chrnorm/deployment-status@releases/v1
-        with:
-          token: "${{ github.token }}"
-          target_url: https://${{ secrets.PROJECT_NAME }}.${{ secrets.CLOUD_REGION }}.gcp.storefrontcloud.io
-          description: Unfortunately, the instance hasn't been updated.
-          state: "failure"
-          deployment_id: ${{ steps.deployment.outputs.deployment_id }}
+  deployment-middleware:
```

Comment:
issue: If we go through Platform's api with deployments, then both requests can be hidden, and made by Platform to Farmer.

Reply:
Initially I thought it was an excellent idea, but on second thought, I don't think that's the case.

If we only make a single request here in the Github Actions CI/CD script and leave the two requests on the Platform/Console side, we will deliver a bad user experience, as a single error may occur in a single deploy of the two and the user will not know exactly what the problem is, as he will only receive an error from Github Actions stating that there was an error. By keeping requests separate, we give more information to the user who can get straight to the issue at hand.

Reply:
Good point.